mdagent/tools/base_tools/analysis_tools/pca_tools.py

```diff
@@ -0,0 +1,370 @@
+            fig_analysis="scree_plot",
+            file_format="png",
+            Sim_id=self.sim_id,
+        )
+        plot_id = self.path_registry.get_fileid(file_name, FileType.FIGURE)
+        path = self.path_registry.ckpt_dir + "/figures/"
+        plt.savefig(path + file_name)  # Save the figure before plt.show()
+        self.path_registry.map_path(plot_id, path + file_name, description=desc)
+        return f"Scree Plot saved as {plot_id} ID\n" + extra_mess
+
+    def make_pc_plots(self):
+        self._make_df()
+        g = sns.PairGrid(
+            self.pc_df,
+            hue="Time (ps)",
+            palette=sns.color_palette("Oranges_d", n_colors=len(self.pc_df)),
+        )
+        g.map(plt.scatter, marker=".")
+        desc = f"PCA Plot comparing the top {self.n_pcs} principal components"
+        file_name = self.path_registry.write_file_name(
+            FileType.FIGURE, fig_analysis="pca", file_format="png", Sim_id=self.sim_id
+        )
+        plot_id = self.path_registry.get_fileid(file_name, FileType.FIGURE)
+        path = self.path_registry.ckpt_dir + "/figures/"
+        plt.savefig(path + file_name)  # Save the figure before plt.show()
+        self.path_registry.map_path(plot_id, path + file_name, description=desc)
+        return f"PCA plots saved as {plot_id} "
+
+    def measure_cosine_convergence(self):
+        if not self.pc:
+            pc_mess = self.get_pc()
+        else:
+            pc_mess = ""
+        if not self.n_pcs:
+            self._get_number_pcs()
+            self._make_transformation()
+
+        pc_messages = []
+        for i in range(self.n_pcs):
+            cc = self._cosine_content(self.transformed, i)
+            pc_messages.append(f"Cosine Content for PC {i+1}-{cc:.3f}\n")
+        cc_message = f"Cosine Content of each PC: {','.join(pc_messages)}"
+        return pc_mess + cc_message
+
+    def run_all(self):
+        self.get_pc()
+        try:
+            scree_plot = self.make_scree_plot()
+        except Exception as e:
+            raise Exception(f"Error during Scree Plot: str({e})")
+        try:
+            pc_plots = self.make_pc_plots()
+        except Exception as e:
+            raise Exception(f"Error during pc plots: str({e})")
+        try:
+            cos_conv = self.measure_cosine_convergence()
+        except Exception as e:
+            raise Exception(f"Error during cosine convergence: str({e})")
+        return f"Analyses done: {scree_plot}, {pc_plots}, {cos_conv}"
+
+
+class PCASchema(BaseModel):
+    trajectory_fileid: str = Field(
+        description="Trajectory File ID of the simulation to be analyzed"
+    )
+    topology_fileid: str = Field(
+        description=("Topology File ID of " "the simulation to be analyzed")
+    )
+    pc_percentage: Optional[float] = Field(
+        95.0, description="Max cumulative percentage of components for analysis"
+    )
+    analysis: str = Field(
+        "all",
+        description=(
```

Comment:
Have you tested this functionality this way? I thought we decided that separate tools would be better than a combo tool with input params to route.. @qcampbel what do you think here?

Reply:
If it works consistently better than the alternative, I'm open to changing it to this approach. I haven't seen this combo approach have higher performance in my experience though.? Whichever way we decide, I'd like to keep things consistent, so we might need to make other changes in the code to reflect it

Reply:
We'll test this in a separate PR.